cmd/translate/main_grpc_integration_test.go

```diff
@@ -14,26 +18,65 @@ import (
 	"google.golang.org/grpc/status"
 )
 
-func createConnection(ctx context.Context, t *testing.T) (*grpc.ClientConn, error) {
-	t.Helper()
+var (
+	transferProtocol string // http or https
+	addr             string
+	port             string
 
-	conn, err := grpc.DialContext(
-		ctx,
-		"localhost:8080",
+	client pb.TranslateServiceClient
+)
+
+func TestMain(m *testing.M) {
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+		main()
+	}()
+
+	// Wait until viper finishes initialization.
+	for viper.GetUint("service.port") == 0 {
```

Comment:
I did not find a better solution to find when viper is finished initializing. Without this waiting `service.port` is 0, e.g. default value.
I tried using the bool flag to indicate that viper values are set in main.go `init()` but couldn't get it to work for some reason

Reply:
There are certain values in the integration tests we can determine on the fly. E.g. free port number for the server to listen on.

We run tests on every save. If the previous test execution is still running, a new one will fail - it cannot listen on the same port which is already in use. But by using a random free port for each test execution we can avoid this conflict. Make the test run more resilient on developers' machines.